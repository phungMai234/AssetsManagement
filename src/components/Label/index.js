import React from 'react';

import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  .default-title {
    font-size: 14px;
    font-weight: 600;
    color: #212529;
    margin-bottom: unset;
  }
  .operator-require {
    color: #ff0000;
    font-size: 14px;
    top: -6px;
  }
`;

const propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element, PropTypes.array]),
  isRequired: PropTypes.bool,
};

const Label = ({ children, isRequired }) => {
  return (
    <Wrapper>
      <label className="default-title">
        {children}
        {isRequired && <sup className="operator-require">*</sup>}
      </label>
    </Wrapper>
  );
};

Label.propTypes = propTypes;

export default Label;
