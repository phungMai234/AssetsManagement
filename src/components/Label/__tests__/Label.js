import React from 'react';

import { render } from '@testing-library/react';

import Label from '../index';

test('should render correctly', () => {
  const container = render(<Label />);
  expect(container.container).toMatchSnapshot();
});

test('should render correctly with required', () => {
  const container = render(<Label isRequired />);
  expect(container.container).toMatchSnapshot();
});
