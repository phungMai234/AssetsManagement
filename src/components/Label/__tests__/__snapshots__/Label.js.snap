// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ kFALVl wrapper-title"
  >
    <label
      class="default-title"
    />
  </div>
</div>
`;

exports[`should render correctly with required 1`] = `
<div>
  <div
    class="sc-bdfBwQ kFALVl wrapper-title"
  >
    <label
      class="default-title"
    >
      <sup
        class="operator-require"
      >
        *
      </sup>
    </label>
  </div>
</div>
`;
